lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -81,19 +107,45 @@ public function get_providers() {
 	public function register_font( $font ) {
 		$font = $this->validate_font( $font );
 		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			$this->webfonts[ $id ] = $font;
+			$slug = $this->get_font_slug( $font );
+
+			if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+				$this->registered_webfonts[ $slug ] = array();
+			}
+
+			$this->registered_webfonts[ $slug ][] = $font;
+		}
+	}
+
+	public function enqueue_font( $font_family ) {
+		$slug = $this->get_font_slug( $font_family );
+
+		if ( isset( $this->enqueued_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is already enqueued.' ), $slug ), '6.0.0' );
+			return false;
+		}
+
+		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is not registered.' ), $slug ), '6.0.0' );
+			return false;
 		}
+
+		$this->enqueued_webfonts[ $slug ] = $this->registered_webfonts[ $slug ];
+		unset( $this->registered_webfonts[ $slug ] );
 	}
 
 	/**
-	 * Get the font ID.
+	 * Get the font slug.
 	 *
 	 * @param array $font The font arguments.
```

Comment:
This method now accepts 2 different data types: `array` or `string`.  
```suggestion
	 * @param array|string $to_convert The value to convert into a slug. Expected as the web font's array or a font-family as a string.
```